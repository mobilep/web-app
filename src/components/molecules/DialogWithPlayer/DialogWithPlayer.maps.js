import { appInfo } from '../../../redux/reducers';

export const mapDispatchToProps = (dispatch) => {
	return {
		onVideoDialogOpen: (isOpen) => dispatch(appInfo.updateVideoPlayState(isOpen)),
	};
};
