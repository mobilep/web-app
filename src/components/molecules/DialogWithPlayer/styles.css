.DialogWithPlayer {
	height: 42vw;
	max-height: 550px;
	padding-top: 45px;
	width: 70vw;
	max-width: 918px;
}

.DialogWithPlayer-header {
	display: none;
	position: absolute;
	top: 0;
	left: 0;
	width: 100%;
	height: 64px;
	border-bottom: 1px solid var(--grey2);
	text-align: center;
	font-size: 16px;
	line-height: 22px;
	color: var(--almostBlack);
	padding-top: 21px;
	font-weight: var(--semiBold);
	z-index: -1;
}

.DialogWithPlayer-playerBase {
	height: 100%;
}

@media (max-width: 768px) {
	.DialogWithPlayer-header {
		display: block;
	}

	.DialogWithPlayer {
		width: 100%;
		max-width: unset;
		height: 100%;
		max-height: unset;
		padding: 0;
		padding-top: 65px;
	}

	button.Dialog-close-button {
		top: 17px;
	}
}
