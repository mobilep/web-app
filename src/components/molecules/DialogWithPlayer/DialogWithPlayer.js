import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Dialog, PlayerBase } from '../../atoms';

import './styles.css';

export default class DialogWithPlayer extends Component {

	static propTypes = {
		fallBackVideoUrl: PropTypes.string,
		onClose: PropTypes.func,
		onVideoDialogOpen: PropTypes.func,
		videoUrl: PropTypes.string,
	}

	static defaultProps = {
		onClose: () => {},
		onVideoDialogOpen: () => {},
	}

	dialogRef = React.createRef();

	componentDidUpdate (prevProps) {
		const { videoUrl } = this.props;
		if (videoUrl && videoUrl !== prevProps.videoUrl) {
			this.dialogRef.current.showModal();
		}

		this.checkVideoDialogOpen();
	}

	checkVideoDialogOpen () {
		const { onVideoDialogOpen } = this.props;
		if (this.dialogRef.current.open) {
			onVideoDialogOpen(true);
		} else onVideoDialogOpen(false);
	}

	static defaultProps = {}

	render () {
		const { fallBackVideoUrl, videoUrl, onClose } = this.props;
		const shouldRenderPlayer = !!videoUrl;
		return (
			<Dialog className={'DialogWithPlayer'} onClose={onClose} ref={this.dialogRef}>
				<div className='DialogWithPlayer-header'>
					Video player
				</div>
				{ shouldRenderPlayer &&
					<PlayerBase
						className='DialogWithPlayer-playerBase'
						streamUrl={videoUrl}
						fallBackStreamUrl={fallBackVideoUrl}
					/>}
			</Dialog>
		);
	}

}
