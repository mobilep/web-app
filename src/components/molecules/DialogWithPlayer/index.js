import React from 'react';
import { connect } from 'react-redux';
import DialogWithPlayer from './DialogWithPlayer';
import { mapDispatchToProps } from './DialogWithPlayer.maps';

export default connect(null, mapDispatchToProps)((props) => <DialogWithPlayer {...props} />);
