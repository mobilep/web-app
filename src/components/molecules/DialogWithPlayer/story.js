import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { storiesOf } from '@storybook/react';
import { VideoTile } from '../../atoms';
import DialogWithPlayer from './DialogWithPlayer';

class _DialogWithPlayer extends Component {

	static propTypes = {
		streamUrl: PropTypes.string,
		thumbnailUrl: PropTypes.string,
	}

	static defaultProps = {}

	handleVideoTileClick = () =>
		this.setState({ streamUrl: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-video/MPEG-DASH/4e0b9b96-2679-446b-827c-11556ca55d11/playlist.mpd' })

	handleDialogClose = () =>
		this.setState({ streamUrl: null })

	render () {
		const { thumbnailUrl } = this.props;
		const { streamUrl } = this.state || {};
		return (
			<div>
				<DialogWithPlayer streamUrl={streamUrl} onClose={this.handleDialogClose} />
				<VideoTile onPlayButtonClick={this.handleVideoTileClick} thumbnailUrl={thumbnailUrl} />
			</div>
		);
	}

}

storiesOf('2. Molecules / DialogWithPlayer', module)
	.add('DialogWithPlayer', () => {
		const thumbnailUrl = 'http://via.placeholder.com/350x150/00423b';
		// eslint-disable-next-line
		return <_DialogWithPlayer {...{ thumbnailUrl }} />;
	});
