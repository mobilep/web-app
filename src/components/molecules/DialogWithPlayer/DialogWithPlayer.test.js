import React from 'react';
import { shallow } from 'enzyme';

import DialogWithPlayer from './DialogWithPlayer';

describe('Molecule - DialogWithPlayer', () => {

	it('Should render', () => {
		const wrapper = shallow(<DialogWithPlayer streamUrl='streamUrl' />);
		expect(wrapper).toMatchSnapshot();
	});

});
