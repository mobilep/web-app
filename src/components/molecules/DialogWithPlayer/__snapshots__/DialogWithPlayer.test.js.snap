// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecule - DialogWithPlayer Should render 1`] = `
<Dialog
  className="DialogWithPlayer"
  closeButton={true}
  onClose={[Function]}
>
  <div
    className="DialogWithPlayer-header"
  >
    Video player
  </div>
</Dialog>
`;
